providers/unleash/pkg/provider.go

```diff
@@ -0,0 +1,168 @@
+	if p.status != of.ReadyState {
+		if p.status == of.NotReadyState {
+			return of.BoolResolutionDetail{
+				Value: defaultValue,
+				ProviderResolutionDetail: of.ProviderResolutionDetail{
+					ResolutionError: of.NewProviderNotReadyResolutionError(providerNotReady),
+					Reason:          of.ErrorReason,
+				},
+			}
+		} else {
+			return of.BoolResolutionDetail{
+				Value: defaultValue,
+				ProviderResolutionDetail: of.ProviderResolutionDetail{
+					ResolutionError: of.NewGeneralResolutionError(generalError),
+					Reason:          of.ErrorReason,
+				},
+			}
+		}
+	}
+
+	res := unleash.IsEnabled(flag, unleash.WithFallback(defaultValue))
+	flagMetadata := map[string]interface{}{
+		"enabled": res,
+	}
+
+	return of.BoolResolutionDetail{
+		Value: res,
+		ProviderResolutionDetail: of.ProviderResolutionDetail{
+			FlagMetadata: flagMetadata,
+		},
+	}
+}
+
+func (p *Provider) FloatEvaluation(ctx context.Context, flag string, defaultValue float64, evalCtx of.FlattenedContext) of.FloatResolutionDetail {
+	res := p.ObjectEvaluation(ctx, flag, defaultValue, evalCtx)
+	return of.FloatResolutionDetail{
+		Value:                    res.Value.(float64),
+		ProviderResolutionDetail: res.ProviderResolutionDetail,
+	}
+}
+
+func (p *Provider) IntEvaluation(ctx context.Context, flag string, defaultValue int64, evalCtx of.FlattenedContext) of.IntResolutionDetail {
+	res := p.ObjectEvaluation(ctx, flag, defaultValue, evalCtx)
+	return of.IntResolutionDetail{
+		Value:                    res.Value.(int64),
+		ProviderResolutionDetail: res.ProviderResolutionDetail,
+	}
+}
+
+func (p *Provider) StringEvaluation(ctx context.Context, flag string, defaultValue string, evalCtx of.FlattenedContext) of.StringResolutionDetail {
+	res := p.ObjectEvaluation(ctx, flag, defaultValue, evalCtx)
+	return of.StringResolutionDetail{
+		Value: fmt.Sprint(res.Value),
+		ProviderResolutionDetail: of.ProviderResolutionDetail{
+			ResolutionError: res.ProviderResolutionDetail.ResolutionError,
+			Reason:          res.ProviderResolutionDetail.Reason,
+			Variant:         res.Variant,
+			FlagMetadata:    res.FlagMetadata,
+		},
+	}
+}
+
+func (p *Provider) ObjectEvaluation(ctx context.Context, flag string, defaultValue interface{}, evalCtx of.FlattenedContext) of.InterfaceResolutionDetail {
+	if p.status != of.ReadyState {
+		if p.status == of.NotReadyState {
+			return of.InterfaceResolutionDetail{
+				Value: defaultValue,
+				ProviderResolutionDetail: of.ProviderResolutionDetail{
+					ResolutionError: of.NewProviderNotReadyResolutionError(providerNotReady),
+					Reason:          of.ErrorReason,
+				},
+			}
+		} else {
+			return of.InterfaceResolutionDetail{
+				Value: defaultValue,
+				ProviderResolutionDetail: of.ProviderResolutionDetail{
+					ResolutionError: of.NewGeneralResolutionError(generalError),
+					Reason:          of.ErrorReason,
+				},
+			}
+		}
+	}
+
+	feature := unleash.GetVariant(flag)
```

Comment:
I don't think we're passing the Unleash context here, which is definitely needed for some toggles to successfully evaluate